server/controllers/userController.js

```diff
@@ -36,6 +43,49 @@ const emailValidator = (req, res, next) => {
     next();
   }
 };
+// check picture is safe or not
+const profileImageValidator = async (imageBuffer) => {
+  const CREDENTIALS = JSON.parse(process.env.GOOGLE_APPLICATION_CREDENTIALS);
+  const CONFIG = {
+    credentials: {
+      private_key: CREDENTIALS.private_key,
+      client_email: CREDENTIALS.client_email,
+    },
+  };
+  const client = new vision.ImageAnnotatorClient(CONFIG);
+
+  const likelihoodToPercentage = {
+    VERY_UNLIKELY: 0,
+    UNLIKELY: 25,
+    POSSIBLE: 50,
+    LIKELY: 75,
+    VERY_LIKELY: 100,
+  };
+  const isUnsafe = (detections) => {
+    const likelihood =
+      detections.adult ||
+      detections.medical ||
+      detections.spoof ||
+      detections.violence ||
+      detections.racy;
```

Comment:
Why not use `Math.max` here?

Reply:
Yeah that also can use.

Reply:
`Math.max` would provide accurate results compared to the current implementation IMO, since the first positive integer > 0 is not guaranteed to always be the largest among the other detections

Reply:
Thanks, got it.

Reply:
plz let me know if something else, I can learn and update.

Reply:
> `Math.max` would provide accurate results compared to the current implementation IMO, since the first positive integer > 0 is not guaranteed to always be the largest among the other detections

Its return Nan.